src/node/platform.cpp

```diff
@@ -134,4 +134,63 @@ void remove_realm_files_from_directory(const std::string &dir_path)
     }
 }
 
+static void remove_directory(const std::string &path)
+{
+    FileSystemRequest exists_req;
+    if(uv_fs_stat(uv_default_loop(), &exists_req, path.c_str(), nullptr) != 0) {
+        if(exists_req.result == UV_ENOENT) {
+            // path doesn't exist, ignore
+            return;
+        } else {
+            throw UVException(static_cast<uv_errno_t>(exists_req.result));
+        }
+    }
+
+    uv_dirent_t dir_entry;
+    FileSystemRequest dir_scan_req;
+    if (uv_fs_scandir(uv_default_loop(), &dir_scan_req, path.c_str(), 0, nullptr) < 0) {
+        throw UVException(static_cast<uv_errno_t>(dir_scan_req.result));
+    }
+
+    while (uv_fs_scandir_next(&dir_scan_req, &dir_entry) != UV_EOF) {
+        std::string dir_entry_path = path + '/' + dir_entry.name;
+        FileSystemRequest delete_req;
+        if (uv_fs_unlink(uv_default_loop(), &delete_req, dir_entry_path.c_str(), nullptr) != 0) {
+            throw UVException(static_cast<uv_errno_t>(delete_req.result));
+        }
+    }
+
+    FileSystemRequest rmdir_req;
+    if (uv_fs_rmdir(uv_default_loop(), &rmdir_req, path.c_str(), nullptr)) {
+        throw UVException(static_cast<uv_errno_t>(rmdir_req.result));
+    }
+}
+
+
+static void remove_file(const std::string &path)
+{
+    FileSystemRequest exists_req;
+    if(uv_fs_stat(uv_default_loop(), &exists_req, path.c_str(), nullptr) != 0) {
+        if(exists_req.result == UV_ENOENT) {
+            // path doesn't exist, ignore
+            return;
+        } else {
+            throw UVException(static_cast<uv_errno_t>(exists_req.result));
+        }
+    }
+
+    FileSystemRequest delete_req;
+    if (uv_fs_unlink(uv_default_loop(), &delete_req, path.c_str(), nullptr) != 0) {
+        throw UVException(static_cast<uv_errno_t>(delete_req.result));
+    }
+}
+
+void remove_realm_files(const std::string &path)
+{
+    remove_file(path);
```

Comment:
I think this logic could be moved to `RealmClass<T>::delete_file` so we at least don't have to duplicate it internally.

Reply:
Good point, done.